pkg/apis/stable/v1alpha1/fleetautoscaler.go

```diff
@@ -0,0 +1,127 @@
+// Copyright 2018 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1alpha1
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+// +genclient
+// +genclient:noStatus
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// FleetAutoScaler is the data structure for a FleetAutoScaler resource
+type FleetAutoScaler struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   FleetAutoScalerSpec   `json:"spec"`
+	Status FleetAutoScalerStatus `json:"status"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// FleetAutoScalerList is a list of Fleet Scaler resources
+type FleetAutoScalerList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+
+	Items []FleetAutoScaler `json:"items"`
+}
+
+// FleetAutoScalerSpec is the spec for a Fleet Scaler
+type FleetAutoScalerSpec struct {
+	FleetName string `json:"fleetName"`
+	// MaxReplicas is the maximum amount of replicas that the fleet may have.
+	// It must be bigger than both MinReplicas and BufferSize
```

Comment:
I think it should be greater than the sum of `MinReplicas` and `BufferSize`

Reply:
Can you share some reasoning behind this?
I don't see it.... :thinking: